### PR TITLE
[PhpUnitBridge] Doc for @expectedDeprecation & new configuration env vars

### PR DESCRIPTION
**Pull request by nicolas-grekas**
Related to https://github.com/symfony/symfony/pull/20255 & https://github.com/symfony/symfony/pull/20256

### CODE REVIEW

File: components/phpunit_bridge.rst [hunk 1]
```diff
@@ -18,9 +18,9 @@ It comes with the following features:
 
 * Displays the stack trace of a deprecation on-demand;
 
-* Provides a ``ClockMock`` helper class for time-sensitive tests.
+* Provides a ``ClockMock`` and ``DnsMock`` helpers class for time or network-sensitive tests.
```
Comment (javiereguiluz) [inline components/phpunit_bridge.rst]: `helpers class` -> `helper classes`

File: components/phpunit_bridge.rst [hunk 2]
```diff
@@ -133,6 +133,28 @@ completely disable the deprecation helper. This is useful to make use of the
...
+``@expectedDeprecation`` annotation that you can use on your test methods,
+alongside with the expected message, given in the same format than for the
+`PHPUnit assertStringMatchesFormat()`_ method). If you expect more than one
+deprecation messages for a give test method, you can use the annotation several
+times::
+
+    /**
+     * @group legacy
+     * @expectedDeprecation This "%s" method is deprecated.
+     * @expectedDeprecation The second argument of the "%s" method is deprecated.
```
Comment (stof) [inline components/phpunit_bridge.rst]: is the order of annotations important ? If they must be in the same order than the triggered deprecation, this should be documented (or changed in the code)

**Review by javiereguiluz — APPROVED** [review]
👍

File: components/phpunit_bridge.rst (hunk 2 above)
Comment (nicolas-grekas) [inline components/phpunit_bridge.rst]: yes it is, open for discussion for sure

Comment (nicolas-grekas) [inline components/phpunit_bridge.rst]: Note added above

File: components/phpunit_bridge.rst [hunk 3]
```diff
@@ -133,6 +133,28 @@ completely disable the deprecation helper. This is useful to make use of the
 rest of features provided by this component without getting errors or messages
 related to deprecations.
 
+Write Assertions about Deprecations
+-----------------------------------
+
+When adding deprecations to your code, you might like writting tests that verify
```
Comment (wouterj) [inline components/phpunit_bridge.rst]: typo: writting -> writing

File: components/phpunit_bridge.rst [hunk 4]
```diff
@@ -133,6 +133,28 @@ completely disable the deprecation helper. This is useful to make use of the
...
+that they are triggered as required. To do so, the bridge provides the
+``@expectedDeprecation`` annotation that you can use on your test methods,
+alongside with the expected message, given in the same format than for the
```
Comment (wouterj) [inline components/phpunit_bridge.rst]: "[...] that you can use on your test methods. It requires you to pass the expected message, [...]"

File: components/phpunit_bridge.rst [hunk 5]
```diff
@@ -133,6 +133,28 @@ completely disable the deprecation helper. This is useful to make use of the
...
+`PHPUnit assertStringMatchesFormat()`_ method. If you expect more than one
+deprecation messages for a given test method, you can use the annotation several
```
Comment (wouterj) [inline components/phpunit_bridge.rst]: "messages" -> "message"

File: components/phpunit_bridge.rst [hunk 6]
```diff
@@ -358,8 +380,19 @@ If you have installed the bridge through Composer, you can run it by calling e.g
...
 
+.. tip::
+
+    Set the ``SYMFONY_PHPUNIT_VERSION`` env var to e.g. ``5.5`` to change the
+    base version of PHPUnit to ``5.5`` instead of the default ``5.3``.
+
+.. tip::
+
+    Set the ``SYMFONY_PHPUNIT_REMOVE`` env var to ``symfony/yaml`` if you need
+    ``prophecy`` but not ``symfony/yaml``.
```
Comment (wouterj) [inline components/phpunit_bridge.rst]: What does "if you need" mean? Am I not able to use Prophecy if I don't set this env var to `symfony/yaml`?

File: components/phpunit_bridge.rst [hunk 7]
```diff
@@ -133,6 +133,28 @@ completely disable the deprecation helper. This is useful to make use of the
...
 related to deprecations.
 
+Write Assertions about Deprecations
+-----------------------------------
+
+When adding deprecations to your code, you might like writting tests that verify
+that they are triggered as required. To do so, the bridge provides the
+``@expectedDeprecation`` annotation that you can use on your test methods,
+alongside with the expected message, given in the same format than for the
+`PHPUnit assertStringMatchesFormat()`_ method. If you expect more than one
```
Comment (wouterj) [inline components/phpunit_bridge.rst]: "PHPUnit's assertStringMatchesFormat() method"

**Review by wouterj — CHANGES_REQUESTED** [review]

File: components/phpunit_bridge.rst (hunk 6 above)
Comment (nicolas-grekas) [inline components/phpunit_bridge.rst]: Yes: as state before, the default is to have prophecy and symfony/yaml removed

**Comment by nicolas-grekas** [comment]
Comments addressed
Status: needs review

**Review by wouterj — APPROVED** [review]

**Comment by wouterj** [comment]
:+1:
status: reviewed

**Comment by xabbuh** [comment]
Thank you @nicolas-grekas.

**Pull request merged**